Source/ORTS.Common/VirtualFileSystem.cs

```diff
@@ -0,0 +1,717 @@
+    /// <br><see cref="File.Exists"/> becomes <see cref="Vfs.FileExists"/>, <see cref="Directory.Exists"/> becomes <see cref="Vfs.DirectoryExists"/>,
+    /// <see cref="Directory.GetFiles(string, string)"/> becomes <see cref="Vfs.GetFiles"/>.</br>
+    /// </summary>
+    public static class Vfs
+    {
+        static VfsNode VfsRoot;
+        static readonly Stack<(string, VfsNode)> Stack = new Stack<(string, VfsNode)>();
+        static readonly ConcurrentDictionary<string, object> OpenArchives = new ConcurrentDictionary<string, object>();
+        static readonly List<string> SupportedArchiveExtensions = new List<string> { ".zip", ".rar" };
+        internal static readonly ConcurrentQueue<string> InitLog = new ConcurrentQueue<string>();
+
+        ///////////////////////////////////////////////////////////////////////////////////////
+        ///
+        /// The configuration file consits of lines of format:
+        /// Drive:\SystemPath\ /MountPoint/
+        ///
+        /// SystemPath - may use "/" or "\", but MountPoint may use "/" only as a directory separator.
+        ///            - with spaces in it must be quoted, e.g. "C:\My Routes\USA85\" /MSTS/ROUTES/USA85/
+        ///            - referring to a directory must end with "/" or "\" to assure that only _below_ dirs and files will be visible _below_ the mount point.
+        ///            - referring to an archive must not end with "/" or "\", e.g. C:\TEMP\MSTS1.2.zip /MSTS/
+        ///            - may refer to an archive subdirectory, e.g.: C:\routes.zip\USA3\ /MSTS/ROUTES/USA3/
+        ///            - may _not_ refer to a single non-archive file, neither a non-archive file within an archive.
+        ///            - is case-insensitive.
+        /// MountPoint - must end with "/" to avoid confusion and to assure that mounting will be done _below_ the given path.
+        ///            - may not contain spaces, may not be quoted.
+        ///            - is case-insensitive, will be converted to all-uppercase internally.
+        ///
+        /// Found subsequent archives are auto-mounted to their respective VFS locations, but nested archives are unsupported.
+        ///
+        /// System.IO.Path.*() functions can be used on virtual paths, but make sure that
+        /// NormalizeVirtualPath() is used on them _afterwards_ in the processing pipeline somewhere,
+        /// so the rest of the code doesn't need to normalize before calling the public functions in this class.
+        /// The exception is the Path.GetFullPath(), because that adds an unwanted drive letter,
+        /// so must never be used on virtual paths.
+        ///
+        ///////////////////////////////////////////////////////////////////////////////////////
+        ///
+        /// Public interfaces
+        /// 
+
+        public const string MstsBasePath = "/MSTS/";
+        public const string ExecutablePath = "/EXECUTABLE/";
```

Comment:
What is this for? I'm feeling like we've skipped the _design_ phase with the architecture of the VFS here.

Reply:
Need to mount the executable "Content" path for being able to access the stored images there. All texture loading got transferred to be done via the VFS, and I didn't want to add a SharedTextureManager.GetNonVfsImage()-like method.